test/UrlParserTest.cpp

```diff
@@ -0,0 +1,71 @@
+// Copyright 2024, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Julian Mundhahs (mundhahj@tf.uni-freiburg.de)
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "util/GTestHelpers.h"
+#include "util/http/HttpUtils.h"
+#include "util/http/UrlParser.h"
+
+using namespace ad_utility;
+
+TEST(UrlParserTest, paramsToMap) {
+  auto parseParams =
+      [](const string& queryString) -> HashMap<std::string, std::string> {
+    boost::urls::url_view url(queryString);
+    return UrlParser::paramsToMap(url.params());
+  };
+  auto HashMapEq = [](const HashMap<std::string, std::string>& hashMap)
+      -> testing::Matcher<HashMap<std::string, std::string>> {
+    return testing::ContainerEq(hashMap);
+  };
+
+  EXPECT_THROW(parseParams("?foo=a&foo=b"), std::runtime_error);
+  // Parameter with key "" is set twice.
+  EXPECT_THROW(parseParams("?&"), std::runtime_error);
+
+  EXPECT_THAT(parseParams("?query=SELECT%20%2A%20WHERE%20%7B%7D"),
+              HashMapEq({{"query", "SELECT * WHERE {}"}}));
+  EXPECT_THAT(parseParams("?cmd=stats"), HashMapEq({{"cmd", "stats"}}));
+  EXPECT_THAT(parseParams("/ping"), HashMapEq({}));
+  EXPECT_THAT(parseParams(""), HashMapEq({}));
+  EXPECT_THAT(parseParams("?"), HashMapEq({{"", ""}}));
```

Comment:
Is that really so that there is an empty key-value pair if you just write `?` Interesting... Maybe add a comment here that this is the behavior of boost and thus probably correct according to some standard.

Reply:
This is a design choice by Boost.URL to make the empty query string distinguishable from no query string. The query string has no standardized semantics.